crate/server/src/config/command_line/db.rs

```diff
@@ -18,22 +23,20 @@ pub const DEFAULT_SQLITE_PATH: &str = "./sqlite-data";
 #[serde(default)]
 pub struct DBConfig {
     /// The database type of the Findex server
-    /// - sqlite: `SQLite`. The data will be stored at the `sqlite_path` directory
-    /// - redis-findex: a Redis database with encrypted data and encrypted indexes thanks to Findex.
-    ///   The Redis url must be provided, as well as the redis-master-password and the redis-findex-label
-    #[clap(
-        long,
-        env("FINDEX_SERVER_DATABASE_TYPE"),
-        value_parser(["sqlite", "redis-findex"]),
-        verbatim_doc_comment
-    )]
-    pub database_type: Option<String>,
+    /// - sqlite: `SQLite`. The data will be stored at the `sqlite_path`
+    ///   directory
+    /// - redis-findex: a Redis database with encrypted data and encrypted
+    ///   indexes thanks to Findex. The Redis url must be provided, as well as
+    ///   the redis-master-password and the redis-findex-label
+    #[clap(long, env("FINDEX_SERVER_DATABASE_TYPE"), verbatim_doc_comment)]
+    pub database_type: Option<DatabaseType>,
 
     /// The url of the database for findex-redis
     #[clap(
         long,
         env = "FINDEX_SERVER_DATABASE_URL",
-        required_if_eq_any([("database_type", "redis-findex")])
+        required_if_eq_any([("database_type", "redis-findex")]),
+        default_value = "redis://localhost:6379"
```

Comment:
I wouldn't use a default value here, but return an error upon failure to parse this information.

Reply:
the idea was to give the format of the URL `redis://XXXX:6379`